Sources/SwiftBuildSupport/SwiftBuildSystem.swift

```diff
@@ -174,6 +174,388 @@ private final class PlanningOperationDelegate: SWBPlanningOperationDelegate, Sen
+                emitInfoAsDiagnostic(info: info)
+            } else {
+                unprocessedDiagnostics.append(info)
+            }
+        case .output(let info):
+            // Grab the taskID to append to buffer-per-task storage
+            guard let taskSignature = info.locationContext2.taskSignature else {
+                // If we cannot find the task signature from the locationContext2,
+                // use deprecated locationContext instead.
+                if let taskID = info.locationContext.taskID,
+                    let taskSignature = buildState.taskSignature(for: taskID) {
+                    buildState.appendToBuffer(taskSignature, data: info.data)
+                }
+
+                return
+            }
+
+            buildState.appendToBuffer(taskSignature, data: info.data)
+        case .taskStarted(let info):
+            try buildState.started(task: info)
+
+            if let commandLineDisplay = info.commandLineDisplayString {
+                self.observabilityScope.emit(info: "\(info.executionDescription)\n\(commandLineDisplay)")
+            } else {
+                self.observabilityScope.emit(info: "\(info.executionDescription)")
+            }
+
+            if self.logLevel.isVerbose {
+                if let commandLineDisplay = info.commandLineDisplayString {
+                    self.outputStream.send("\(info.executionDescription)\n\(commandLineDisplay)")
+                } else {
+                    self.outputStream.send("\(info.executionDescription)")
+                }
+            }
+            let targetInfo = try buildState.target(for: info)
+            buildSystem.delegate?.buildSystem(buildSystem, willStartCommand: BuildSystemCommand(info, targetInfo: targetInfo))
+            buildSystem.delegate?.buildSystem(buildSystem, didStartCommand: BuildSystemCommand(info, targetInfo: targetInfo))
+        case .taskComplete(let info):
+            let startedInfo = try buildState.completed(task: info)
+
+            // If we've captured the compiler output with formatted diagnostics, emit them.
+//            if let buffer = buildState.dataBuffer(for: startedInfo) {
+            emitDiagnosticCompilerOutput(startedInfo)
+//            }
+
+            if info.result != .success {
+                self.observabilityScope.emit(severity: .error, message: "\(startedInfo.ruleInfo) failed with a nonzero exit code. Command line: \(startedInfo.commandLineDisplayString ?? "<no command line>")")
+            }
+            let targetInfo = try buildState.target(for: startedInfo)
+            buildSystem.delegate?.buildSystem(buildSystem, didFinishCommand: BuildSystemCommand(startedInfo, targetInfo: targetInfo))
+            if let targetName = targetInfo?.targetName {
+                serializedDiagnosticPathsByTargetName[targetName, default: []].append(contentsOf: startedInfo.serializedDiagnosticsPaths.compactMap {
+                    try? Basics.AbsolutePath(validating: $0.pathString)
```

Comment:
Just let this throw, if this isn't an absolute path something has gone very wrong.